### ISSUE TITLE
Write a notebook to explore the idea of using Bayesian Network for implementing joint distribution of agent features and also sampling from the model.

### ISSUE THREAD

**Issue by jiqicn** [covid19ABM/comma#35]

**Comment by jiqicn** [comment]
We firstly need to check if the variable pairs given in the table are independent or not in our dataset. The idea: https://rstudio-pubs-static.s3.amazonaws.com/107408_cef5c04db4d24678a55b25c490eba58f.html

**Comment by jiqicn** [comment]
The notebook is now available [here](https://github.com/covid19ABM/comma/blob/bn_exploration/notebooks/bn_exploration.ipynb). 

@n400peanuts, since we both have the notebooks ready, we can find a time and discuss the pros/cons of each.

**Comment by n400peanuts** [comment]
UPDATE: sorry I had the other PR open! I made a mistake -- haha. This is the right PR I was mentioning!

-----

Ehy @jiqicn,

Thanks a lot for this!! I like this approach a lot! especially because I like your approach of considering one variable independent (e.g., age) and then build dependency for the other variables from there. The library seems simple enough as well and it doesn't require variables to have the same number of levels. I'd say let's go for it!!

Great idea, this I think 100% solves our sampling from a representative distribution problem. I would only see whether we can use all our features, but for this we may want to check with Kristina first if we can prune (as a possibility) those variables whereby she doesn't have all the people represented -- it would make the task less computationally challenging.

great job!!!

**Comment by jiqicn** [comment]
> Great idea, this I think 100% solves our sampling from a representative distribution problem. I would only see whether we can use all our features, but for this we may want to check with Kristina first if we can prune (as a possibility) those variables whereby she doesn't have all the people represented -- it would make the task less computationally challenging.

Agree. We should indeed do some feature selection. I'm sure we can do some pruning, not only by population coverage but also by other criteria, e.g. independence test.

Given the feature selection result, we may want to first design the Bayesian network(s) so that we know what CPDs we need from Kristina. We may also want to communicate the network structure with Kristina to see if she also agrees with it.

**Comment by n400peanuts** [comment]
@jiqicn Let's discuss this together with Kristina today. How would you feel about explaining to her the Bayesian Framework approach to reproduce the distribution? Nothing too complicated, but only the general idea. Since this was your (brilliant) idea and I wish Kristina to know that, it would be great if you could lead the discussion! 

No pressure of course, only if you wish to do it and it requires only minimal preparation!

**Comment by jiqicn** [comment]
@n400peanuts I can lead the discussion, no problem. And let's see how she feels.

**Comment by jiqicn** [comment]
Bayesian Network plot.
![image](https://github.com/covid19ABM/comma/assets/92043159/b8a33424-caa3-47a5-8573-6d3984016b9a)
